packages/components/close-button/src/CloseButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { ButtonStylesProps } from '@spark-ui/button/src/Button.styles'
+import { Icon } from '@spark-ui/icon'
+import { DeleteFill } from '@spark-ui/icons/dist/icons/DeleteFill'
+import React, { forwardRef, PropsWithChildren, useCallback, useRef } from 'react'
+
+import { closeButtonIconStyles, closeButtonStyles } from './CloseButton.styles'
+
+export interface CloseButtonProps
+  extends PropsWithChildren<Omit<React.ButtonHTMLAttributes<HTMLButtonElement>, 'disabled'>>,
+    ButtonStylesProps {
+  disabled?: boolean
+  onClick?: React.MouseEventHandler<HTMLSpanElement>
+  label?: string
+}
+
+export const CloseButton = forwardRef<HTMLSpanElement, PropsWithChildren<CloseButtonProps>>(
+  (
+    { disabled, onClick, className, label = 'close', children = <DeleteFill />, ...props },
+    forwardedRef
+  ) => {
+    const buttonRef = useRef<HTMLDivElement>(null)
+    const onCloseHandler = useCallback(
+      (event: React.MouseEvent<HTMLButtonElement>) => {
+        event.stopPropagation()
+        onClick && onClick(event)
+      },
+      [onClick]
+    )
+
+    const onKeyUpHandler = useCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
+      ;['Enter'].includes(event.key) && buttonRef.current?.click()
+    }, [])
+
+    return (
+      <span
+        ref={forwardedRef}
+        className={closeButtonStyles({
+          cursor: disabled ? 'disabled' : 'pointer',
+          className,
+        })}
+        onClick={onCloseHandler}
+        {...props}
+      >
+        <div
+          ref={buttonRef}
+          role="button"
+          {...(disabled ? { 'aria-disabled': disabled } : { tabIndex: 0 })}
```

Comment:
If it were a normal `button` the `disabled` attribute already manages this behaviour.

Reply:
yes, its not for adding it or not, its because it is only required to be placed only if its disabled, so i dnt want to add more html properties than the required ones